src/cryptographic_primitives/proofs/low_degree_exponent_interpolation.rs

```diff
@@ -0,0 +1,215 @@
+use std::fmt;
+
+use derivative::Derivative;
+use thiserror::Error;
+
+use crate::cryptographic_primitives::hashing::traits::Hash;
+use crate::cryptographic_primitives::proofs::ProofError;
+use crate::cryptographic_primitives::Polynomial;
+use crate::elliptic::curves::traits::ECPoint;
+
+/// Claims that there's polynomial `w(x)` of degree `deg(w) <= degree`, and
+/// `forall i. x[i] = g[i] * alpha[i]` (and the prover knows `w(x)`)
+#[derive(Derivative)]
+#[derivative(Clone(bound = "P: Clone, P::Scalar: Clone"))]
+#[derivative(Debug(bound = "P: fmt::Debug, P::Scalar: fmt::Debug"))]
+pub struct LdeiStatement<P: ECPoint> {
+    pub alpha: Vec<P::Scalar>,
+    pub g: Vec<P>,
+    pub x: Vec<P>,
+    pub degree: u16,
+}
+
+#[derive(Derivative)]
+#[derivative(Clone(bound = "P: Clone, P::Scalar: Clone"))]
+#[derivative(Debug(bound = "P: fmt::Debug, P::Scalar: fmt::Debug"))]
+pub struct LdeiProof<P: ECPoint> {
+    pub a: Vec<P>,
+    pub e: P::Scalar,
+    pub z: Polynomial<P>,
+}
+
+impl<P> LdeiProof<P>
+where
+    P: ECPoint + Clone + PartialEq,
+    P::Scalar: Clone + PartialEq,
+{
+    /// Constructs [LdeiStatement] and proves it correctness
+    ///
+    /// ## Protocol
+    ///
+    /// The prover samples `u(X) ← Z_q[X]` with `deg(u) ≤ d` and computes `a_i = g_i^u(alpha_i)`
+    /// for all `i ∈ [m]`, in addition to `e = H(g_1,...,g_m,x_1,...,x_m,a_1,...,a_m)`, and
+    /// `z(X) = u(X) − e · w(X)`. The proof is `(a_1,...,a_m,e,z)`.
+    #[allow(clippy::many_single_char_names)]
+    pub fn prove<H>(
+        w: &Polynomial<P>,
+        alpha: Vec<P::Scalar>,
+        g: Vec<P>,
+    ) -> Result<(LdeiStatement<P>, LdeiProof<P>), LdeiProveError>
+    where
+        H: Hash,
+    {
+        if alpha.len() != g.len() {
```

Comment:
the prover should check correctness of the statement (same for the verifier): 
I see that a check that g.len()>=w.degree

Reply:
Added the check